File: backend/src/main/java/com/horain/repository/EvalBacklogRepository.java
package com.horain.repository;

import com.horain.model.EvalBacklog;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;
import java.util.UUID;

/**
 * JPA repository for eval backlog (candidates for Promptfoo promotion).
 */
public interface EvalBacklogRepository extends JpaRepository<EvalBacklog, UUID> {

    List<EvalBacklog> findByStatus(String status);

    List<EvalBacklog> findByTurnId(UUID turnId);
}

File: backend/src/main/kotlin/com/horain/repository/EvalBacklogRepository.kt
package com.horain.repository

import com.horain.model.EvalBacklog
import org.springframework.data.jpa.repository.JpaRepository
import java.util.UUID

/**
 * JPA repository for eval backlog (candidates for Promptfoo promotion).
 */
interface EvalBacklogRepository : JpaRepository<EvalBacklog, UUID> {

    fun findByStatus(status: String): List<EvalBacklog>

    fun findByTurnId(turnId: UUID): List<EvalBacklog>
}
